src/lib.rs

```diff
@@ -1225,6 +1113,114 @@ impl LdapClient {
     }
 }
 
+pub struct Stream<'a, S, A> {
```

Comment:
This is still only a "stream like object" (just like in `ldap3`). We want to return an object that implements the [`futures::stream::Stream`](https://docs.rs/futures/latest/futures/stream/trait.Stream.html) trait. I confess that I haven't ever implemented this myself, but taking inspiration from [`page_turner::mt::PageTurner::pages_ahead`](https://docs.rs/page-turner/latest/page_turner/mt/trait.PageTurner.html#method.pages_ahead) it would seem like [`futures::stream::try_unfold`](https://docs.rs/futures/latest/futures/stream/fn.try_unfold.html) would be able to do most of the work. Probably you can more or less just wrap the code you just wrote inside that.

What's the point you may ask? With `Stream` implemented we can then bring in [`StreamExt`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html) trait which allows us to use `map`, `filter` etc all the iterator like stuff on the stream.

Reply:
Implemented Stream trait for Stream.

Reply:
Yeah hopefully we'll get language support at some point. Perhaps just blocking for the abandoning would be an acceptable compromise. Honestly I don't know how angry the LDAP/AD servers get if you just leave a search dangling?